packages/server/src/product/product.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CreateProductDto } from './dto/create-product.dto';
+import { UpdateProductDto } from './dto/update-product.dto';
+import { Product } from './product.interface';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectModel('Product') private readonly productModel: Model<Product>,
+  ) {}
+
+  findAll() {
+    return this.getAll();
+  }
+
+  find(id: string) {
+    return this.get(id);
+  }
+
+  create(createProductDto: CreateProductDto) {
+    const product: Product = {
+      title: createProductDto.title,
+      price: createProductDto.price,
+      description: createProductDto.description,
+      imgUrl: createProductDto.imgUrl,
+    };
+    return this.addProduct(product);
+  }
+
+  update(id: string, updateProductDto: UpdateProductDto) {
+    const product: Product = {
+      title: updateProductDto.title,
+      price: updateProductDto.price,
+      description: updateProductDto.description,
+      imgUrl: updateProductDto.imgUrl,
+    };
+    return this.updateProduct(id, product);
+  }
+
+  delete(id: string) {
+    return this.deleteProduct(id);
+  }
+
+  private async getAll() {
+    const result = await this.productModel.find({});
+    return result;
+  }
+
+  private async get(id: string) {
+    const result = await this.productModel.findById(id);
+    return result;
+  }
+
+  private async addProduct(product: Product) {
+    const newProduct = new this.productModel(product);
```

Comment:
what if we get error here ? Incase product doesn't contain all required fields etc?